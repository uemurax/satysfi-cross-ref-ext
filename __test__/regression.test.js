const shell = require("shelljs");
const fs = require("fs");
const tmp = require("tmp");

const { toMatchPdfSnapshot } = require("jest-pdf-snapshot");
expect.extend({ toMatchPdfSnapshot });

shell.cd("__test__");

function compilerOutput(src) {
    const tmpFile = tmp.fileSync();
    const saty = shell.exec(`satysfi ${src} -o ${tmpFile.name} -b`);
    const code = saty.code;
    const stdout = saty
          .exec("awk '/evaluating texts .../{flag=1;next}/evaluation done/{flag=0}flag'")
          .stdout;
    return {
        code,
        stdout,
    }
}

const baseDir = "base"

afterAll(() => {
    shell.rm(`${baseDir}/*.satysfi-aux`);
});

test("SATySFi is installed", () => {
    expect(shell.exec("satysfi --version").code).toBe(0);
});

describe("Snapshot tests (compiler outputs)", () => {
    const filenames = [
        "type.saty",
    ];
    for (const filename of filenames) {
        test(`compiler output of ${filename}`, () => {
            const res = compilerOutput(`${baseDir}/${filename}`);
            expect(res.code).toBe(0);
            expect(res.stdout).toMatchSnapshot();
        });
    }
});
