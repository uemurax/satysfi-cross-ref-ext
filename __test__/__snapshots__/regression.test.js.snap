// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests (compiler outputs) compiler output of type.saty 1`] = `
"[INFO] n1 passed
[INFO] n2 passed
[INFO] s1 passed
[INFO] l1 passed
"
`;
